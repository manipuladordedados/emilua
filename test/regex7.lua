-- From GAWK's rationale[1] (check test regex6.lua too):
--
-- > regular expression matching with FPAT can match the null string, and the
-- > non-matching intervening characters function as the separators
--
-- [1] https://www.gnu.org/software/gawk/manual/html_node/FS-versus-FPAT.html

local regex = require 'regex'

local re = regex.new{
    pattern = '[^,]*',
    grammar = 'extended'
}

local s = {
    ',Arnold,,MyTown,MyState,12345-6789,',
    'Robbins,Arnold,,MyTown,MyState,12345-6789,',
    ',Arnold,,MyTown,MyState,12345-6789,USA'
}

for _, si in ipairs(s) do
    for k, v in ipairs(regex.patsplit(re, si)) do
        print(k, v)
    end
end

print '--'

for _, si in ipairs(s) do
    si = byte_span.append(si)
    for k, v in ipairs(regex.patsplit(re, si)) do
        print(k, v)
    end
end
