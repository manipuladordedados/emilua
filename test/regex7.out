1	
2	Arnold
3	
4	MyTown
5	MyState
6	12345-6789
7	
1	Robbins
2	Arnold
3	
4	MyTown
5	MyState
6	12345-6789
7	
1	
2	Arnold
3	
4	MyTown
5	MyState
6	12345-6789
7	USA
--
1	
2	Arnold
3	
4	MyTown
5	MyState
6	12345-6789
7	
1	Robbins
2	Arnold
3	
4	MyTown
5	MyState
6	12345-6789
7	
1	
2	Arnold
3	
4	MyTown
5	MyState
6	12345-6789
7	USA
